pygtracker/pygtracker.py

```diff
@@ -80,7 +82,69 @@ def generate_course_statistics(self, course_ids):
                 median: float
                 3rd-quantile: float
         """
-        return None
+
+        course_list = ["511", "522"]  # = list(self.courses.course_id.unique())
+
+        # check input type
+        if type(course_ids) != list:
+            raise TypeError("course_ids should be a list of str")
+
+        for i in range(len(course_ids)):
+            if type(course_ids[i]) == str:
+                pass
+            elif type(course_ids[i]) == int:
```

Comment:
You shouldn't assume that the code here is int and try to convert it to str. To make it simple, just assume it is a str. What if later they decide to input it as "DSC511"?